pkg/util/loop/loop.go

```diff
@@ -160,37 +179,36 @@ func (loop *Device) AttachFromFile(image *os.File, mode int, number *int) error
 				return nil
 			}
 			syscall.Close(loopFd)
+			continue
 		} else {
 			_, _, esys := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetFd, image.Fd())
 			if esys != 0 {
 				syscall.Close(loopFd)
 				continue
 			}
-			break
 		}
-	}
 
-	if _, _, err := syscall.Syscall(syscall.SYS_FCNTL, uintptr(loopFd), syscall.F_SETFD, syscall.FD_CLOEXEC); err != 0 {
-		return fmt.Errorf("failed to set close-on-exec on loop device %s: %s", path, err.Error())
-	}
+		if _, _, err := syscall.Syscall(syscall.SYS_FCNTL, uintptr(loopFd), syscall.F_SETFD, syscall.FD_CLOEXEC); err != 0 {
+			return fmt.Errorf("failed to set close-on-exec on loop device %s: %s", path, err.Error())
+		}
 
-	maxRetries := 5
-	for i := 0; i < maxRetries; i++ {
 		if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetStatus64, uintptr(unsafe.Pointer(loop.Info))); err != 0 {
-			if err == syscall.EAGAIN && i < maxRetries-1 {
-				// with changes introduces in https://github.com/torvalds/linux/commit/5db470e229e22b7eda6e23b5566e532c96fb5bc3
-				// loop_set_status() can temporarily fail with EAGAIN -> sleep and try again
-				// (cf. https://github.com/karelzak/util-linux/blob/dab1303287b7ebe30b57ccc78591070dad0a85ea/lib/loopdev.c#L1355)
-				time.Sleep(250 * time.Millisecond)
```

Comment:
I am somewhat concerned that we are losing the retry sleepy after getting EAGAIN and also to some degree that we are not trying the ioctl again right away. The EAGAIN errno specifically means that we are being asked to try again (https://pubs.opengroup.org/onlinepubs/000095399/basedefs/errno.h.html), and from what I can tell it does seem to be normal to implement this retry with some waiting (e.g. see losetup code which is probably the de facto reference implementation for handling loop device ioctl's: https://github.com/util-linux/util-linux/blob/2401078c9cf9ddec105e44af074cf2fcbd4ef153/sys-utils/losetup.c#L576-L579 in which they use a 200ms wait between retries for up to 64 attempts).

I suppose it is reasonable here given that we are happy to accept using any loop device we find we can get our hands on to come back around to this one after trying others, but I'm concerned that in some situations it may mean that we come back around to retry the same device after significantly less time has passed.